pytorch_blade/src/compiler/backends/backend_input_outputs.cpp

```diff
@@ -0,0 +1,136 @@
+// Copyright 2022 The BladeDISC Authors. All rights reserved.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "backend_input_outputs.h"
+#include "compiler/jit/shape_type_spec.h"
+#include "compiler/jit/tool_funcs.h"
+
+namespace torch {
+namespace blade {
+namespace backends {
+
+bool DynamicRanges::Validate(int inp_nums) {
```

Comment:
It's better to keep a pull request small and clean. it seems that `DynamicRanges`  is not necessary for this refactor task.

ref: https://developers.google.com/blockly/guides/modify/contribute/write_a_good_pr